resolve/config.go

```diff
@@ -64,7 +64,12 @@ func (o regexpOverrideSpec) matches(imp ImportSpec, lang string) bool {
 }
 
 func (o regexpOverrideSpec) resolveRegexpDep(imp ImportSpec) label.Label {
-
+	// If NumSubexp() returns 0, then there are no captures in the regex,
```

Comment:
Now that I thought about this more, I realized that adding a capturing group without the intention of capturing it is probably somewhat common (`(a|b)` is simply less verbose than `(?:a|b)`). 

Maybe looking for back references in the replacement string is a better signal? What do you think?

Reply:
Sorry for the delay, last week was very busy for me.  I should get back around to looking at this a bit today.

Reply:
I looked into it a bit more and I agree with your statement.  Backreferences, or `regex.Expand` usage via "$" is a better signal.

Reply:
Thanks, don't worry, we're all busy :⁠-⁠)